draft-ietf-jsonpath-base.md

```diff
@@ -1466,6 +1469,12 @@ independent of any query argument the JSONPath query is applied to.
 | `NodesType`          | Nodelists                       |
 {: #tbl-types title="Function extension type system"}
 
+A function is typed in accordance with its result.
+
+* Functions which return `ValueType` are `value-function-expr`
+* Functions which return `LogicalType` are `logical-function-expr`
+* Functions which return `NodesType` are `nodes-function-expr`
```

Comment:
> This text feels clunky and I'm open to improvements. I considered adding it to the table above, but I couldn't figure out how to work it in.

How about the following?
```suggestion
Function expressions are annotated with their result type in the ABNF via the following synonyms:

* `value-function-expr` is a `function-expression` that returns `ValueType`,
* `logical-function-expr` is a `function-expression` that returns `LogicalType`,
* `nodes-function-expr` is a `function-expression` that returns `NodesType`.

~~~~ abnf
value-function-expr = function-expr
logical-function-expr = function-expr
nodes-function-expr = function-expr
~~~~
```

If the suggestion is adopted, the ABNF definitions would need deleting where they currently occur.